addons/mail/tests/test_mail_activity.py

```diff
@@ -1,10 +1,101 @@
 from freezegun import freeze_time
 
 from odoo import exceptions
+from odoo import fields
+from odoo.addons.mail.tests.common import MailCommon
 from odoo.addons.mail.tests.common_activity import ActivityScheduleCase
 from odoo.tests import tagged, HttpCase
 
 
+@tagged('post_install', '-at_install')
+class TestMailActivitySearch(MailCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+    def test_search_date_done_direct(self):
+        """ Test searching directly on mail.activity for done activities. """
+
+        self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': self.partner_admin.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Active To-Do',
+            'date_deadline': '2025-09-01',
+
+        })
+
+        done_activity1 = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': self.partner_admin.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do 1',
+            'date_deadline': '2025-09-02',
+        })
+        done_activity1.action_done()
+
+        done_activity2 = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': self.partner_admin.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do 2',
+            'date_deadline': '2025-09-03'
+        })
+        done_activity2.action_done()
+
+        self.assertEqual(self.env['mail.activity'].search_count([]), 1)
+        self.assertEqual(self.env['mail.activity'].search_count([('date_done', '=', fields.Date.today())]), 2)
+        self.assertEqual(self.env['mail.activity'].search_count([
+            '&',
+            ('date_done', '=', fields.Date.today()),
+            ('date_deadline', '=', '2025-09-03'),
+        ]), 1)
+
+    def test_search_date_done_related_field(self):
+        """ Test searching through a related field (res.partner's activity_ids). """
+        partner_with_active_activity = self.env['res.partner'].create({'name': 'Test Partner1 for Activities'})
+        partner_with_done_activity = self.env['res.partner'].create({'name': 'Test Partner2 for Activities'})
```

Comment:
Maybe we can also have an archived record to ensure we only disable "active_test" for activities, not for the original record.

And then maybe have a test with explicit `active_test=True` and `active_test=False`